tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -123,103 +183,193 @@ int is_valid_NetworkInterface(IotNetworkInterface_t *netif) {
+
 /* Creates a Connection Info and assigns memory accordingly. */
-IotHttpsConnectionInfo_t * newConnectionInfo() {
-  IotHttpsConnectionInfo_t * pConnInfo = safeMalloc(sizeof(IotHttpsConnectionInfo_t));
+IotHttpsConnectionInfo_t * allocate_IotConnectionInfo() {
+  IotHttpsConnectionInfo_t * pConnInfo =
+    safeMalloc(sizeof(IotHttpsConnectionInfo_t));
   if(pConnInfo) {
-    pConnInfo->pNetworkInterface = newNetworkInterface();
-    /* Function _createHttpsConnection checks if pConnInfo->addressLen is less than
-    IOT_HTTPS_MAX_HOST_NAME_LENGTH. The +1 is added to cover all cases. */
-    __CPROVER_assume(pConnInfo->addressLen >= 0 && pConnInfo->addressLen <= IOT_HTTPS_MAX_HOST_NAME_LENGTH + 1);
+    pConnInfo->pNetworkInterface = allocate_NetworkInterface();
     pConnInfo->pAddress = safeMalloc(pConnInfo->addressLen);
-    /* Function _createHttpsConnection checks if pConnInfo->alpnProtocolsLen is less than
-    IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH. The +1 is added to cover all cases. */
-    __CPROVER_assume(pConnInfo->alpnProtocolsLen >= 0 && pConnInfo->alpnProtocolsLen <= IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH + 1);
     pConnInfo->pAlpnProtocols = safeMalloc(pConnInfo->alpnProtocolsLen);
-    pConnInfo->pCaCert = malloc(sizeof(uint32_t));
-    pConnInfo->pClientCert = malloc(sizeof(uint32_t));
-    pConnInfo->pPrivateKey = malloc(sizeof(uint32_t));
+    pConnInfo->pCaCert = safeMalloc(sizeof(uint32_t));
+    pConnInfo->pClientCert = safeMalloc(sizeof(uint32_t));
+    pConnInfo->pPrivateKey = safeMalloc(sizeof(uint32_t));
     pConnInfo->userBuffer.pBuffer = safeMalloc(sizeof(_connHandle_t));
   }
   return pConnInfo;
 }
 
+int is_valid_IotConnectionInfo(IotHttpsConnectionInfo_t *pConnInfo) {
+  return
+    pConnInfo->pCaCert &&
+    pConnInfo->pClientCert &&
+    pConnInfo->pPrivateKey &&
+    pConnInfo->userBuffer.pBuffer &&
+    pConnInfo->pNetworkInterface &&
+    is_valid_NetworkInterface(pConnInfo->pNetworkInterface);
+}
+
+/****************************************************************
+ * IotHttpsConnectionHandle constructor
+ ****************************************************************/
+
 /* Creates a Connection Handle and assigns memory accordingly. */
-IotHttpsConnectionHandle_t newIotConnectionHandle () {
-  IotHttpsConnectionHandle_t pConnectionHandle = safeMalloc(sizeof(_connHandle_t));
+IotHttpsConnectionHandle_t allocate_IotConnectionHandle () {
+  IotHttpsConnectionHandle_t pConnectionHandle =
+    safeMalloc(sizeof(_connHandle_t));
   if(pConnectionHandle) {
-    pConnectionHandle->pNetworkConnection = safeMalloc(sizeof(_connHandle_t));
-    pConnectionHandle->pNetworkInterface = newNetworkInterface();
+    // network connection just points to an allocated memory object
+    pConnectionHandle->pNetworkConnection = safeMalloc(1);
+    pConnectionHandle->pNetworkInterface = allocate_NetworkInterface();
+    // ???: Should reqQ initialization be an assumption?
```

Comment:
It is a lot easier to make it empty. 
It sounds complicated to make it either empty or have items? That would require setting the links (nodes) correctly to wrap back around?

Reply:
This constructor creates empty queues, but items are installed in the test harness itself.